components/table/Table.tsx

```diff
@@ -375,7 +380,7 @@ function Table<RecordType extends object = any>(props: TableProps<RecordType>) {
     (innerColumns: ColumnsType<RecordType>): ColumnsType<RecordType> => {
       return transformTitleColumns(
         transformSelectionColumns(transformFilterColumns(transformSorterColumns(innerColumns))),
-      );
+      ).filter(c => !c.responsive || c.responsive.filter((r: Breakpoint) => matched.has(r)).length);
```

Comment:
Move this to the begining.

```tsx
const mergedColumn = React.useMemo(() => { .... }, [columns, children, ...]);
```

And we can now remove `children` parse to columns in `useSorter` & `useSelection`.

Reply:
I've moved the memo up top as you suggested. For some reason the responsive display has broken though. I've checked that `mergedColumn` returns the filtered set. Any ideas?